src/Stratis.FederatedPeg.Features.FederationGateway/Notifications/BlockObserver.cs

```diff
@@ -20,22 +23,39 @@ internal class BlockObserver : SignalObserver<ChainedHeaderBlock>
 
         private readonly IDepositExtractor depositExtractor;
 
+        private readonly IBlockStore blockStore;
+
+        private readonly IFullNode fullNode;
+
+        private readonly uint minimumDepositConfirmations;
+
+        private readonly ConcurrentChain chain;
+
         /// <summary>
         /// Initialize the block observer with the wallet manager and the cross chain monitor.
         /// </summary>
         /// <param name="walletSyncManager">The wallet sync manager to pass new incoming blocks to.</param>
         /// <param name="crossChainTransactionMonitor">The crosschain transaction monitor to pass new incoming blocks to.</param>
+        /// <param name="depositExtractor">The component used to extract the deposits from the blocks appearing on chain.</param>
+        /// <param name="federationGatewaySettings">The settings used to run this federation node.</param>
+        /// <param name="fullNode">Full node used to get rewind the chain.</param>
         public BlockObserver(IFederationWalletSyncManager walletSyncManager, 
                              ICrossChainTransactionMonitor crossChainTransactionMonitor,
-                             IDepositExtractor depositExtractor)
+                             IDepositExtractor depositExtractor,
+                             IFederationGatewaySettings federationGatewaySettings,
+                             IFullNode fullNode)
         {
             Guard.NotNull(walletSyncManager, nameof(walletSyncManager));
             Guard.NotNull(crossChainTransactionMonitor, nameof(crossChainTransactionMonitor));
             Guard.NotNull(depositExtractor, nameof(depositExtractor));
+            Guard.NotNull(federationGatewaySettings, nameof(federationGatewaySettings));
+            Guard.NotNull(fullNode, nameof(fullNode));
 
             this.walletSyncManager = walletSyncManager;
             this.crossChainTransactionMonitor = crossChainTransactionMonitor;
             this.depositExtractor = depositExtractor;
+            this.minimumDepositConfirmations = federationGatewaySettings.MinimumDepositConfirmations;
+            this.chain = fullNode.NodeService<ConcurrentChain>();

```

Comment:
I'm not sure if we want to introduce `ConcurrentChain` into new code, considering it is supposed to be removed at some point. `ChainedHeaderTree is probably the place to look, although I don't know offhand if it has an exactly equivalent .GetBlock(int)` method.

Reply:
OK I'll have a look, thanks for that

Reply:
Dan says use concurrent chain for now, especially if we are only concerned about stuff that can't be reorged. So I'll leave it like that for the moment.